doc/userdoc/contribute/styleguide/styleguide.rst

```diff
@@ -0,0 +1,634 @@
+
+
+Sometimes in the documentation you want to show a complete function call, as in ``nest.Create("iaf_psc_apha")``.
+In these cases, the link cannot be used.
+
+
+See `the Sphinx documentation on referencing Python objects
+<https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects>`_ for more
+information.
+
+
+reStructuredText text formatting
+--------------------------------
+
+Underlines for headings
+~~~~~~~~~~~~~~~~~~~~~~~
+
+reStructuredText uses several types of underline markers for headings. It is
+important that the length of the underline is exactly as long as the words
+in the heading.
+
+In general, we try to follow the pattern:
+
+* First heading: ``===``
+* Second heading: ``---``
+* Third heading: ``~~~``
+* Fourth heading: ``^^^``
+
+"Double quotes"
+~~~~~~~~~~~~~~~
+
+We use double quotes for strings in code, for example,  ``nest.Create("iaf_psc_alpha")``. This applies to
+reStructuredText files as well as Python and C++ code. This rule is based on PEP 257, which (only) dictates the use of
+double quotes in triple quoted strings; for consistency, double quotes are used throughout the codebase.
+
+Double or single quotes should not be used to emphasize important concepts in the text.
+
+.. _sec_dbltick:
+
+\``Double backticks\``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Use double backticks for all code and command related terms, such as function call examples, paths, variables, and parameters.
+In addition, meta and special characters (such as the ampersand ``&``) should also be written in double backticks.
+
+For example::
+
+    ``nest.Create("iaf_psc_alpha")``
+
+    ``/path/to/source/file.rst``
+
+    "The key ``rule`` in the connectivity specification dictionary ``conn_spec`` . . . "
+
+
+\**Strong emphasis\** vs \*emphasis\*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Avoid using strong emphasis (boldface) in texts. If you want text to stand out use an appropriate :ref:`admonition <sec_admonition>`.
+
+Use the plain emphasis (italics) to signify context differences *within* a text.
+
+
+NumPy style docstrings
+----------------------
+
+In PyNEST code, we follow most of the rules for NumPy style docstrings as
+`explained in the NumPy style guide <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+
+However, we use different formatting marks than what is stated in their guide. See section on
+:ref:`double back ticks <sec_dbltick>`.
+
+If you are contributing to the :doc:`PyNEST API <example_template>`, make sure you carefully read the NumPy guide, along
+with this one.
+
+
+Bibliography style
+------------------
+
+The reStructuredText bibliography style is used throughout the documentation so that links
+are autogenerated and a consistent format is used.
+
+For in-text citations, we use the reStructuredText numeric style ``[1]_``.
+
+For example:
+
+    The following example is based on Smith [1]_.
+    Sanders et al. [2]_ contains a technically detailed example.
```

Comment:
I am a bit confused here. Is this meant as an example of how to enter references in RST? Why are the references then rendered orange in Readthedocs? On the other hand, if this is to show the end result, why are the underscores shown in the rendered version?

Reply:
Yah, I see the confusion. I modified the text to make it a rendered example